? socket.patch
? window_SetWindowStyleFlag.patch
? build/vs8
? include/wx/socket.h-
? src/common/socket.cpp-
? src/msw/gsocket.cpp-
Index: src/msw/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/window.cpp,v
retrieving revision 1.674
diff -u -r1.674 window.cpp
--- src/msw/window.cpp	2006/03/04 18:17:58	1.674
+++ src/msw/window.cpp	2006/03/07 20:20:57
@@ -1155,6 +1155,7 @@
     if ( !GetHwnd() )
         return;
 
+    bool swp = false;                     // Should call SetWindowPos?
     WXDWORD exstyle, exstyleOld;
     long style = MSWGetStyle(flags, &exstyle),
          styleOld = MSWGetStyle(flagsOld, &exstyleOld);
@@ -1170,17 +1171,32 @@
         styleReal |= style;
 
         ::SetWindowLong(GetHwnd(), GWL_STYLE, styleReal);
+
+        // If any of the style changes changed any of the frame styles:
+        // MSDN: SetWindowLong:
+        //       Certain window data is cached, so changes you make using 
+        //       SetWindowLong will not take effect until you call the 
+        //       SetWindowPos function. Specifically, if you change any of 
+        //       the frame styles, you must call SetWindowPos with the 
+        //       SWP_FRAMECHANGED flag for the cache to be updated properly.
+
+        swp = ( ( styleOld ^ style ) & ( WS_BORDER | WS_THICKFRAME | WS_CAPTION | WS_DLGFRAME | WS_MAXIMIZEBOX | WS_MINIMIZEBOX | WS_SYSMENU ) ) != 0;
     }
 
     // and the extended style
+    long exstyleReal = ::GetWindowLong(GetHwnd(), GWL_EXSTYLE);
+
     if ( exstyle != exstyleOld )
     {
-        long exstyleReal = ::GetWindowLong(GetHwnd(), GWL_EXSTYLE);
         exstyleReal &= ~exstyleOld;
         exstyleReal |= exstyle;
 
         ::SetWindowLong(GetHwnd(), GWL_EXSTYLE, exstyleReal);
+        swp = true;
+    }
 
+    if ( swp )
+    {
         // we must call SetWindowPos() to flush the cached extended style and
         // also to make the change to wxSTAY_ON_TOP style take effect: just
         // setting the style simply doesn't work
@@ -1188,7 +1204,7 @@
                              exstyleReal & WS_EX_TOPMOST ? HWND_TOPMOST
                                                          : HWND_NOTOPMOST,
                              0, 0, 0, 0,
-                             SWP_NOMOVE | SWP_NOSIZE) )
+                             SWP_NOMOVE | SWP_NOSIZE | SWP_FRAMECHANGED) )
         {
             wxLogLastError(_T("SetWindowPos"));
         }

 	  	 
